index.ts

```diff
@@ -8,91 +8,43 @@ import {
 import env from "./env.json";
 import MyClient from "./MyClient";
 
-type NestedToken = Omit<TokenByIdResult, "nestingChildTokens"> & {
-  nestingChildTokens: NestedToken[];
-};
+(async function main() {
+  console.log("Started", env);
+  await MyClient.init();
+  console.log("connected");
+  console.time("magic1");
 
-const MAX_BUNDLE_DEPTH = 5;
 
-async function getTopmostParent(
-  this: Sdk,
-  collectionId: number,
-  tokenId: number
-): Promise<NestedToken> {
-  const parent = await this.tokens.parent({ collectionId, tokenId });
-  if (parent)
-    return getTopmostParent.call(this, parent.collectionId, parent.tokenId);
-  const token = await this.tokens.get_new({ collectionId, tokenId });
-  if (!token) throw new Error("Token was not found");
-  return { ...token, nestingChildTokens: [] };
-}
+  
+  for(let i = 0; i < 1; i++) { // update 'i < 1' accordingly if you need to get more than 1 token
+    const res = await MyClient.magic(2, 2);
+    console.dir(res, { depth: 5 }); // to fix because this step returns undefined
+    console.log('The token has been created');
+    console.log('-------------------------')
+  }
+  // !!! uncomment required block of the code below to create tokens with another depth
 
-async function getChildTokens(
-  this: Sdk,
-  childrenResult: TokenChildrenResult,
-  depth = 0
-): Promise<NestedToken[]> {
-  if (depth > MAX_BUNDLE_DEPTH)
-    throw new Error(
-      "Recursion depth exceeded, aborting NestedToken generation"
-    );
-  if (!childrenResult?.children?.length) return [];
-  // fetch every child token using recursion in parallel
-  const childTokens = await Promise.all<NestedToken>(
-    childrenResult.children.map(async (children: TokenChildrenArguments) => {
-      const childToken = await this.tokens.get_new(children);
-      if (!childToken)
-        throw new Error("Token not found during search for nested childs");
-      const nestedChildToken = childToken as NestedToken;
-      const childChildren = await this.tokens.children(children);
-      if (childChildren?.children?.length)
-        nestedChildToken.nestingChildTokens = await getChildTokens.call(
-          this,
-          childChildren,
-          depth + 1
-        );
-      return nestedChildToken;
-    })
-  );
+  // for(let i = 0; i < 1; i++) { // update 'i < 1' accordingly if you need to get more than 1 token
```

Comment:
put depth/width in env as well and remove this commented part

Reply:
done